Platformio/HAL/HardwareAbstract.hpp

```diff
@@ -46,6 +56,8 @@ class HardwareAbstract {
     Notification<batteryStatus> mBatteryNotification;
 
   private:
+    std::vector<std::function<void()>> wifi_scan_start_cb;
```

Comment:
You can wrap these in the Notification Class 
Well I guess I don't know if we can make the template without data but my guess is it would look like 
Notification<> wifi_scan_start;

The one that has data for sure can do this. 
Notification<std::shared_ptr<std::vector<WifiInfo>> wifi_scan_done;